drivers/counter/counter_native_posix.c

```diff
@@ -105,12 +141,18 @@ static int ctr_set_alarm(const struct device *dev, uint8_t chan_id,
 		return -ENOTSUP;
 	}
 
+	if (is_top_set && alarm_cfg->ticks > top.ticks) {
```

Comment:
shouldnt this check be done after offseting if `!(alarm_cfg->flags & COUNTER_ALARM_CFG_ABSOLUTE)` ?

Reply:
Agreed, fixed.